### PR TITLE
Fixed nested field predicate pushdown

### PR DESCRIPTION
**Pull request by davidrabinowitz**

### CODE REVIEW

**Review by isha97 — APPROVED** [review]

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkFilterUtils.java [hunk 1]
```diff
@@ -338,6 +338,8 @@ static String escape(String value) {
   }
 
   static String quote(String value) {
-    return "`" + value + "`";
+    return value.contains(".")
+        ? Arrays.stream(value.split("\\.")).collect(Collectors.joining("`.`", "`", "`"))
```
Comment (satybald) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkFilterUtils.java:342]: could we add a comment to describe better what's going on here?

Comment (davidrabinowitz) [reply spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkFilterUtils.java:346]: Added comment, thanks for the suggestion

**Pull request merged**